mgmt: ec_host_cmd: add logging system

niedzwiecki-dawid opened the pull request:
Add a native way to log Host Command communication.

Introduce a few logging levels. It allows adjusting the amount of log data depending on user needs - debug/normal usage.

Allow users to install a custom logging function.

niedzwiecki-dawid committed:
mgmt: ec_host_cmd: add logging system

Add a native way to log Host Command communication.

Use Zephyr logging system to do it. Use debug and normal levels.

Signed-off-by: Dawid Niedzwiecki <dawidn@google.com>

fabiobaltieri commented on subsys/mgmt/ec_host_cmd/Kconfig.logging:
```diff
@@ -0,0 +1,20 @@
+	int "Max buffer to store request or response"
+	default 129
+	help
+	  Set maximum size of buffer to store request/response in hex. The amount of
```
"in bytes"? maybe it's obvious unsure...

semihalf-niedzwiecki-dawid replied:
Done.

fabiobaltieri commented on subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c:
```diff
@@ -196,15 +271,15 @@ FUNC_NORETURN static void ec_host_cmd_thread(void *hc_handle, void *arg2, void *
 		status = verify_rx(rx);
 		if (status != EC_HOST_CMD_SUCCESS) {
-			send_status_response(hc->backend, tx, status);
+			ec_host_cmd_send_response(status, &args);
```
unrelated change?

semihalf-niedzwiecki-dawid replied:
It is related. I want to use one function to send all responses. It allows adding logging only to that function. I've created a separated commit for it.

fabiobaltieri commented on subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c:
```diff
@@ -213,7 +288,7 @@ FUNC_NORETURN static void ec_host_cmd_thread(void *hc_handle, void *arg2, void *
 		/* No handler in this image for requested command */
 		if (found_handler == NULL) {
-			send_status_response(hc->backend, tx, EC_HOST_CMD_INVALID_COMMAND);
+			ec_host_cmd_send_response(EC_HOST_CMD_INVALID_COMMAND, &args);
```
same

semihalf-niedzwiecki-dawid replied:
Same as above.

niedzwiecki-dawid committed:
mgmt: ec_host_cmd: use one command to send response

Use one common function to send Host Command response. It allows
handling all response types within one function.

Signed-off-by: Dawid Niedzwiecki <dawidn@google.com>

semihalf-niedzwiecki-dawid commented on subsys/mgmt/ec_host_cmd/Kconfig.logging:
```diff
@@ -0,0 +1,20 @@
+	  Set maximum size of buffer to store request/response in hex. The amount of
+	  data sent via host command can be big and it would require a big buffer to
+	  print that in one piece. To print "n" number of data, a buffer with size =
+	  "2*n + 1" is required.
```
Done.

semihalf-niedzwiecki-dawid commented on subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c:
```diff
@@ -161,10 +181,26 @@ int ec_host_cmd_send_response(enum ec_host_cmd_status status,
+
+		bin2hex(args->output_buf, (hex_buf_len - 1) / 2, hex_buf, hex_buf_len);
+		if (needed_hex_buf_len > CONFIG_EC_HOST_CMD_LOG_BUF_MAX) {
+			ec_host_cmd_print("HC resp:%s..", hex_buf);
```
No secret meaning :-)

semihalf-niedzwiecki-dawid commented on subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c:
```diff
@@ -174,6 +210,45 @@ int ec_host_cmd_send_response(enum ec_host_cmd_status status,
+static void ec_host_cmd_log_request(const uint8_t *rx_buf)
+{
+	static uint16_t prev_cmd;
+	const struct ec_host_cmd_request_header *const rx_header = (void *)rx_buf;
```
Done.

keith-zephyr commented on subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c:
```diff
@@ -161,10 +161,20 @@ int ec_host_cmd_send_response(enum ec_host_cmd_status status,
 		return status;
 	}
 
+#ifdef CONFIG_EC_HOST_CMD_LOG_DBG_BUFFERS
```
Non-blocking - can you use `if (IS_ENABLED(CONFG...))` here?

semihalf-niedzwiecki-dawid replied:
Yes, done.

fabiobaltieri commented on subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c:
```diff
@@ -174,6 +185,40 @@ int ec_host_cmd_send_response(enum ec_host_cmd_status status,
+			 */
+			char dbg_raw[sizeof(dbg_fmt)];
+
+			sprintf(dbg_raw, dbg_fmt, rx_header->cmd_id, rx_header->cmd_ver);
```
Can you use snprintf? Just in case.

semihalf-niedzwiecki-dawid replied:
Sure, makes sense. Done.
